laravel/app/Providers/AppServiceProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Providers;
+
+use Illuminate\Support\ServiceProvider;
+
+class AppServiceProvider extends ServiceProvider
+{
+    /**
+     * Bootstrap any application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+    }
+
+    /**
+     * Register any application services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->bindInterfaces();
+    }
+
+    /**
+     * Bind interfaces to their implementations for dependency injection
+     *
+     * @return void
+     */
+    private function bindInterfaces(): void
```

Comment:
This is where you define which implementation the dependency injector will use when we pass an interface into a functions constructor.